docs/redirectless.md

```diff
@@ -0,0 +1,98 @@
+There are two ways to create a configuration to initiate the redirectless flow:
+1. The configuration object contians `onAuthorizationSuccess` and the required attributes (e.g. `client_id`, `response_type`, etc.) used by the internal authorization request function.
+1. The configuration object contians `onAuthorizationRequest` and `onAuthorizationSuccess` functions.
+
+Most of deployments should use the first option. The second option is there for advanced use-cases.
+### `onAuthorizationSuccess` function
+This callback function is going to be called during authorization request. It doesn't have any arguments and it's expected to return a JavaScript `Promise` which completes the authorization request call to PingFederate.
+
+Here's an example:
+```javascript
+var config = {
+  onAuthorizationRequest: function () {
+    var url = 'https://localhost:9031/as/authorization.oauth2?client_id=test&response_type=token&response_mode=pi.flow'
+    var options = {
+      method: 'GET',
+      credentials: 'include'
+    }
+    return fetch(url, options);
+  }
+}
+```
+The `credentials: 'include'` Options attribute is required to make sure PF's session cookie is handled correctly by the browser.
+
+### `onAuthorizationRequest` function
+This callback function is designed to return the result of the transcation the webpage containing the authentication api widget. The protocol response is passed to this function as the first argument when called by the authentication API widget.
+
+Here's an example: 
+```js
+var config = {
+  onAuthorizationSuccess: function (response) {
+    console.log(response.access_token);
+  }
+};
+```
+
+### Supported attributes
+The following attributes are supported by the internal obAuthorizationRequest function: `client_id`, `response_type`, `code_challenge`, `code_challenge_method`, `redirect_uri`, `scope`, `state`, `idp`, `pfidpadapterid`, `access_token_manager_id`, `aud`, `nonce`, `prompt`, `acr_values`, `max_age`, `login_hint`, `ui_locales`, `id_token_hint`, `claims_locales`. 
+
+The `client_id` and `response_type` attributes are required if `onAuthorizationRequest` is not present. 
+
+The present key value pairs will be appended to the authorization request URL, the value will be url encoded and arrays will be concatinated and url encoded into one string.
+
+## Configuration object examples
+### OAuth 2.0 implicit
+This code snipet demonstrates a simple OAuth 2.0 configuration used by the Authentication API widget redirectless flow.
+```javascript
+var config = {
+  client_id: 'test',
+  response_type: 'token',
+  scopes: ['a', 'b', 'c'],
+  onAuthorizationSuccess: function (response) {
+    console.log(response.access_token);
+  }
+};
+```
+### OpenID Connect
+This code snipet demonstrates a simple OpenID Connect configuration used by the Authentication API widget redirectless flow.
+```javascript
+var config = {
+  scope: ['openid', 'profile', 'email', 'address', 'phone'],
+  nonce: '3f05dd88-3e97-496f-ba04-50e36e7ee1a5',
```

Comment:
We don't want to hardcode the nonce here right?

Reply:
Correct